docker/Dockerfile

```diff
@@ -37,11 +51,16 @@ RUN scripts/install_tools.sh
 RUN go generate && go build -tags production -o kjudge cmd/kjudge/main.go
 
 # Stage 3: Create awesome output image
-FROM ubuntu:jammy
+FROM base-ubuntu
 
-RUN apt-get update && DEBIAN_FRONTEND="noninteractive" TZ="Asia/Ho_Chi_Minh" apt-get install -y \
-    build-essential openjdk-11-jdk-headless fp-compiler python3.6 cgroup-lite python2.7 rustc golang libcap-dev # Compilers  \
-    openssl # For HTTPS support
+# Python PPA
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
+    --mount=type=cache,target=/var/lib/apt,sharing=locked \
+    apt-get update && apt-get install -y software-properties-common && add-apt-repository ppa:deadsnakes/ppa && \
+    apt-get update && DEBIAN_FRONTEND="noninteractive" TZ="Asia/Ho_Chi_Minh" apt-get install -y \
+    build-essential openjdk-11-jdk-headless fp-compiler cgroup-lite \
+    python3.6 python2.7 rustc golang libcap-dev \
```

Comment:
[sidenote] Grr `apt` why are you deducing `python3.6` to be some `libcasa` bs...